db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -12276,6 +12295,18 @@ windowDefinition(WindowList wl) throws StandardException :
     }
 }
 
+Long
+asOfClause() throws StandardException :
+{
+    Long txId = null;
+}
+{
+    LOOKAHEAD({(getToken(1).kind == AS && getToken(2).kind == OF)}) <AS> <OF> txId = exactNumber()
```

Comment:
Is this lookahead necessary?

Reply:
I am afraid it is, in order to resolve ambiguity for statements like:
`SELECT * FROM T AS x`
`SELECT * FROM T AS OF <something>`
Without the lookahead, I think the first rule (line 10934) will be matched causing an error. Therefor the lookahead is relevant.